core/app/models/spree/stock_location.rb

```diff
@@ -12,8 +12,8 @@ class StockLocation < Spree::Base
     scope :active, -> { where(active: true) }
     scope :order_default, -> { order(default: :desc, name: :asc) }
 
-    after_create :create_stock_items, :if => "self.propagate_all_variants?"
     after_save :ensure_one_default
+    after_commit :create_stock_items, on: :create, if: "self.try :propagate_all_variants?"
```

Comment:
Could have the seeds destroy the migrated data and seed fresh records if seeds are run.